The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12032551156/job/33544665184) [(plain)](https://github.com/rust-lang/rust/commit/8b69be3c63ff672f558700f495937eabf34ffd69/checks/33544665184/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.899 Building wheels for collected packages: reuse
#13 2.900   Building wheel for reuse (pyproject.toml): started
#13 3.147   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.148   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.148   Stored in directory: /tmp/pip-ephem-wheel-cache-oy3o1fab/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.151 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.556 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.557 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.092 Collecting virtualenv
#13 4.092 Collecting virtualenv
#13 4.139   Downloading virtualenv-20.28.0-py3-none-any.whl (4.3 MB)
#13 4.298      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 27.4 MB/s eta 0:00:00
#13 4.355 Collecting platformdirs<5,>=3.9.1
#13 4.362   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.401 Collecting filelock<4,>=3.12.2
#13 4.430 Collecting distlib<1,>=0.3.7
#13 4.437   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.448      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 47.4 MB/s eta 0:00:00
#13 4.448      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 47.4 MB/s eta 0:00:00
#13 4.529 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.710 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.28.0
#13 DONE 4.8s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      300992 kB
DirectMap2M:     9136128 kB
DirectMap1G:     9437184 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2024-10-16/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 30.12s
##[endgroup]
fmt check
Diff in /checkout/library/std/src/thread/local.rs:259:
     /// thread_local! {
     ///     pub static STATIC: String = String::from("I am");
-    /// 
+    ///
     /// assert_eq!(
     /// assert_eq!(
     ///     STATIC.with(|original_value| format!("{original_value} initialized")),
     ///     "I am initialized",
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/std/src/collections/hash/map/tests.rs" "/checkout/library/std/src/collections/hash/set.rs" "/checkout/library/std/src/collections/hash/mod.rs" "/checkout/library/std/src/collections/hash/map.rs" "/checkout/library/std/src/collections/mod.rs" "/checkout/library/std/src/rt.rs" "/checkout/library/std/src/macros.rs" "/checkout/library/std/src/panicking.rs" "/checkout/library/std/src/fs/tests.rs" "/checkout/library/std/src/hash/random.rs" "/checkout/library/std/src/hash/mod.rs" "/checkout/library/std/src/f128/tests.rs" "/checkout/library/std/src/io/error/repr_unpacked.rs" "/checkout/library/std/src/io/error/tests.rs" "/checkout/library/std/src/io/error/repr_bitpacked.rs" "/checkout/library/std/src/io/cursor.rs" "/checkout/library/std/src/io/error.rs" "/checkout/library/std/src/io/cursor/tests.rs" "/checkout/library/std/src/io/copy/tests.rs" "/checkout/library/std/src/io/prelude.rs" "/checkout/library/std/src/io/stdio.rs" "/checkout/library/std/src/io/impls/tests.rs" "/checkout/library/std/src/io/buffered/bufreader.rs" "/checkout/library/std/src/io/buffered/bufwriter.rs" "/checkout/library/std/src/io/buffered/linewritershim.rs" "/checkout/library/std/src/io/buffered/bufreader/buffer.rs" "/checkout/library/std/src/io/buffered/tests.rs" "/checkout/library/std/src/io/buffered/linewriter.rs" "/checkout/library/std/src/io/buffered/mod.rs" "/checkout/library/std/src/io/copy.rs" "/checkout/library/std/src/io/impls.rs" "/checkout/library/std/src/io/tests.rs" "/checkout/library/std/src/io/util/tests.rs" "/checkout/library/std/src/io/stdio/tests.rs" "/checkout/library/std/src/io/mod.rs" "/checkout/library/std/src/io/util.rs" "/checkout/library/std/src/f32.rs" "/checkout/library/std/src/thread/current.rs" "/checkout/library/std/src/thread/local.rs" "/checkout/library/std/src/thread/local/tests.rs" "/checkout/library/std/src/thread/local/dynamic_tests.rs" "/checkout/library/std/src/thread/scoped.rs" "/checkout/library/std/src/thread/spawnhook.rs" "/checkout/library/std/src/thread/tests.rs" "/checkout/library/std/src/thread/mod.rs" "/checkout/library/std/src/os/fuchsia/fs.rs" "/checkout/library/std/src/os/fuchsia/raw.rs" "/checkout/library/std/src/os/fuchsia/mod.rs" "/checkout/library/std/src/os/ios/mod.rs" "/checkout/library/std/src/os/macos/mod.rs" "/checkout/library/std/src/os/solid/io.rs" "/checkout/library/std/src/os/solid/ffi.rs" "/checkout/library/std/src/os/solid/mod.rs" "/checkout/library/std/src/os/wasi/fs.rs" "/checkout/library/std/src/os/wasi/net/mod.rs" "/checkout/library/std/src/os/wasi/io/fd/tests.rs" "/checkout/library/std/src/os/wasi/io/raw.rs" "/checkout/library/std/src/os/wasi/io/fd.rs" "/checkout/library/std/src/os/wasi/io/mod.rs" "/checkout/library/std/src/os/wasi/ffi.rs" "/checkout/library/std/src/os/wasi/mod.rs" "/checkout/library/std/src/os/haiku/fs.rs" "/checkout/library/std/src/os/haiku/raw.rs" "/checkout/library/std/src/collections/hash/set/tests.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Tue Nov 26 14:27:44 UTC 2024
  network time: Tue, 26 Nov 2024 14:27:44 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/std/src/thread/local.rs
+++ b/library/std/src/thread/local.rs
@@ -252,6 +252,19 @@ impl<T: 'static> LocalKey<T> {
     /// This function will `panic!()` if the key currently has its
     /// destructor running, and it **may** panic if the destructor has
     /// previously been run for this thread.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// thread_local! {
+    ///     pub static STATIC: String = String::from("I am");
+    /// }
+    ///
+    /// assert_eq!(
+    ///     STATIC.with(|original_value| format!("{original_value} initialized")),
+    ///     "I am initialized",
+    /// );
+    /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn with<F, R>(&'static self, f: F) -> R
     where
@@ -273,6 +286,19 @@ impl<T: 'static> LocalKey<T> {
     ///
     /// This function will still `panic!()` if the key is uninitialized and the
     /// key's initializer panics.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// thread_local! {
+    ///     pub static STATIC: String = String::from("I am");
+    /// }
+    ///
+    /// assert_eq!(
+    ///     STATIC.try_with(|original_value| format!("{original_value} initialized")),
+    ///     Ok(String::from("I am initialized")),
+    /// );
+    /// ```
     #[stable(feature = "thread_local_try_with", since = "1.26.0")]
     #[inline]
     pub fn try_with<F, R>(&'static self, f: F) -> Result<R, AccessError>
@@ -452,7 +478,7 @@ impl<T: 'static> LocalKey<RefCell<T>> {
     /// Panics if the key currently has its destructor running,
     /// and it **may** panic if the destructor has previously been run for this thread.
     ///
-    /// # Example
+    /// # Examples
     ///
     /// ```
     /// use std::cell::RefCell;
@@ -483,7 +509,7 @@ impl<T: 'static> LocalKey<RefCell<T>> {
     /// Panics if the key currently has its destructor running,
     /// and it **may** panic if the destructor has previously been run for this thread.
     ///
-    /// # Example
+    /// # Examples
     ///
     /// ```
     /// use std::cell::RefCell;

